tests/Library/Vanilla/Utility/HtmlUtilsTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * @author Todd Burry <todd@vanillaforums.com>
+ * @copyright 2009-2020 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace VanillaTests\Library\Vanilla\Utility;
+
+use PHPUnit\Framework\TestCase;
+use Vanilla\Utility\HtmlUtils;
+
+/**
+ * Tests for the `HtmlUtils` class.
+ */
+final class HtmlUtilsTest extends TestCase {
+    /**
+     * Tests for `HtmlUtils::classNames()`
+     *
+     * @param array $args
+     * @param string $expected
+     * @dataProvider provideClassNameTests
+     */
+    public function testClassNames(array $args, string $expected): void {
+        $actual = HtmlUtils::classNames(...$args);
+        $this->assertSame($expected, $actual);
+    }
+
+    /**
+     * Provide tests for `testClassNames()`.
+     * @return array
+     */
+    public function provideClassNameTests(): array {
+        $r = [
+            [[''], ''],
+            [['foo', 'bar'], 'foo bar'],
+            [['foo', null, 'bar'], 'foo bar'],
+            [['foo', '', 'bar'], 'foo bar'],
+        ];
+        return $r;
+    }
+
+    /**
+     * Formatting tags when the argument isn't supplied is a notice.
+     */
+    public function testFormatTagsInvalidArgNotice(): void {
+        $actual = @HtmlUtils::formatTags('Hello <0/>');
+        $this->assertSame('Hello ', $actual);
+
+        $this->expectNotice();
+        $this->expectExceptionMessage('<0/>');
```

Comment:
I'm not seeing how this actually works. It doesn't look like the the notice I see thrown above. Maybe this should be in the `expectNotice()`?

Reply:
I think it works because notices are exceptions in PHPUnit, however it is a typo so I'll change. As an FYI, when I changed I tried changing the message in the test and the test failed so it is getting called properly.